tests/cmd-runner

```diff
@@ -0,0 +1,173 @@
+running commands the driver can record various attributes about
+the runner process.  So far this is just memory usage.
+
+There's a BaseClientDriverTest test that implements the driver part
+of the tests.  These should read like normal functional/integration tests.
+
+This script (cmd-runner) implements the runner.  It listens for commands
+from the driver and runs the commands.
+
+The driver and runner communicate via a basic text protocol.  Commands
+are line separated with arguments separated by spaces.  Commands
+are sent to the runner via stdin.
+
+On success, the runner response through stdout with "OK".
+
+Here's an example::
+
+    +--------+                                     +-------+
+    | Driver |                                     |Runner |
+    +--------+                                     +-------+
+
+         create_client s3
+         -------------------------------------------->
+               OK
+         <-------------------------------------------
+
+         make_aws_request 100 ec2 describe_instances
+         -------------------------------------------->
+               OK
+         <-------------------------------------------
+
+         stream_s3_upload bucket key /tmp/foo.txt
+         -------------------------------------------->
+               OK
+         <-------------------------------------------
+
+         exit
+         -------------------------------------------->
+               OK
+         <-------------------------------------------
+
+"""
+import botocore.session
+import sys
+
+
+class CommandRunner(object):
+    def __init__(self):
+        self._session = botocore.session.get_session()
+        self._clients = []
+
+    def run(self):
+        while True:
+            line = sys.stdin.readline()
+            parts = line.strip().split()
+            if not parts:
+                break
+            elif parts[0] == 'exit':
+                sys.stdout.write('OK\n')
+                sys.stdout.flush()
+                break
+            else:
+                getattr(self, '_do_%s' % parts[0])(parts[1:])
+                sys.stdout.write('OK\n')
+                sys.stdout.flush()
+
+    def _do_create_client(self, args):
+        # The args provided by the user map directly to the args
+        # passed to create_client.  At the least you need
+        # to provide the service name.
+        self._clients.append(self._session.create_client(*args))
+
+    def _do_create_multiple_clients(self, args):
+        # Args:
+        # * num_clients - Number of clients to create in a loop
+        # * client_args - Variable number of args to pass to create_client
+        num_clients = args[0]
+        client_args = args[1:]
+        for _ in range(int(num_clients)):
+            self._clients.append(self._session.create_client(*client_args))
+
+    def _do_free_clients(self, args):
+        # Frees the memory associated with all clients.
+        self._clients = []
+
+    def _do_gc_collect(self, args):
```

Comment:
Not used, I'll remove.  I was initially using this because botocore has some cyclic references we can clean up, but we're not using it now so I'll remove it.